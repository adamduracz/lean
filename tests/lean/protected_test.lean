namespace nat
  check induction_on      -- ERROR
  check rec_on            -- ERROR
  check nat.induction_on
  check lt.rec_on         -- OK
  check nat.lt.rec_on
  namespace lt
    check rec_on          -- ERROR
    check lt.rec_on
  end lt
end nat
