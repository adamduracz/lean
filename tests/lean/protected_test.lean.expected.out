protected_test.lean:2:8: error: unknown identifier 'induction_on'
protected_test.lean:3:8: error: unknown identifier 'rec_on'
nat.induction_on : ∀ (n : ℕ), ?C 0 → (∀ (a : ℕ), ?C a → ?C (succ a)) → ?C n
lt.rec_on : ?a < ?a_1 → ?C (succ ?a) → (∀ {b : ℕ}, ?a < b → ?C b → ?C (succ b)) → ?C ?a_1
lt.rec_on : ?a < ?a_1 → ?C (succ ?a) → (∀ {b : ℕ}, ?a < b → ?C b → ?C (succ b)) → ?C ?a_1
protected_test.lean:8:10: error: unknown identifier 'rec_on'
lt.rec_on : ?a < ?a_1 → ?C (succ ?a) → (∀ {b : ℕ}, ?a < b → ?C b → ?C (succ b)) → ?C ?a_1
